user_tools/src/spark_rapids_pytools/rapids/qualification.py

```diff
@@ -597,12 +607,36 @@ def __build_global_report_summary(self,
             # No need to run saving estimator or process the data frames.
             return QualificationSummary(comments=[self.__generate_mc_types_conversion_report])
 
+        apps_pruned_df, prune_notes = self.__remap_columns_and_prune(all_apps)
+        recommended_apps = self.__get_recommended_apps(apps_pruned_df)
+        # if the gpu_reshape_type is set to JOB then, then we should ignore recommended apps
+        speedups_irrelevant_flag = self.__recommendation_is_non_standard()
+        reshaped_notes = self.__generate_cluster_shape_report()
+        report_comments = [prune_notes] if prune_notes else []
+        if reshaped_notes:
+            report_comments.append(reshaped_notes)
+
+        pricing_config = self.ctxt.platform.configs.get_value_silent('pricing')
         reshape_col = self.ctxt.get_value('local', 'output', 'processDFProps',
                                           'clusterShapeCols', 'columnName')
         speed_recommendation_col = self.ctxt.get_value('local', 'output', 'speedupRecommendColumn')
-        apps_pruned_df, prune_notes = self.__remap_columns_and_prune(all_apps)
-        recommended_apps = self.__get_recommended_apps(apps_pruned_df)
         apps_reshaped_df, per_row_flag = self.__apply_gpu_cluster_reshape(apps_pruned_df)
+        # OnPrem platform doesn't have pricing information. We do not calculate cost savings for
+        # OnPrem platform.
+        if pricing_config is None:
+            # Write to qualification_summary.csv file
+            if apps_reshaped_df is not None:
+                apps_reshaped_df = apps_reshaped_df.drop(columns=['Estimated Job Frequency (monthly)'])
+                self.logger.info('Generating GPU Estimated Speedup as %s', csv_out)
+                apps_reshaped_df.to_csv(csv_out, float_format='%.2f')
+
+            return QualificationSummary(comments=report_comments,
+                                        all_apps=apps_pruned_df,
+                                        recommended_apps=recommended_apps,
+                                        df_result=apps_reshaped_df,
+                                        irrelevant_speedups=speedups_irrelevant_flag,
+                                        sections_generators=[self.__generate_mc_types_conversion_report])
+
```

Comment:
this part has a lot of repeated code that is going to be hard to maintain moving forward.
We may need to think what is the best way to reformat that block so we won't have multiple places when we are extending the code or debugging an issue.

Reply:
Thanks for the feedback. I have refactored the code to remove repeated code. PTAL.